app/frontend/src/features/auth/Logout.tsx

```diff
@@ -1,14 +1,17 @@
 import React, { useEffect } from "react";
+import { useQueryClient } from "react-query";
 import { Redirect } from "react-router-dom";
 import { loginRoute } from "../../AppRoutes";
 import { useAuthContext } from "./AuthProvider";
 
 export default function Logout() {
   const { authState, authActions } = useAuthContext();
+  const queryClient = useQueryClient();
 
   useEffect(() => {
     if (authState.authenticated) {
       authActions.logout();
+      queryClient.resetQueries("ping");
```

Comment:
Actually, should we reset everything for security? It's fine to leave users since anyone authenticated can access them, but there's some private stuff like messages which would still be cached.

Reply:
Yeah that's probably a good shout, since we cache a copy of it in localStorage so probably best to do that so the persist plugin code will delete it from localStorage too.